webpack/components/AnsibleRolesSwitcher/components/AnsibleRoleInputs.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const AnsibleRoleInputs = props => {
+  const { role, idx, resourceName } = props;
+
+  const fieldName = attr =>
+    `${resourceName}[${resourceName}_ansible_roles_attributes][${idx}][${attr}]`;
+
+  const attrName = modelName => `${modelName}AnsibleRoleId`;
+
+  const idField = (
+    <input
+      name={fieldName('id')}
+      value={role[attrName(resourceName)]}
+      type="hidden"
+    />
+  );
+  return (
+    <React.Fragment>
+      {role[attrName(resourceName)] ? idField : null}
+      <input
+        name={fieldName('ansible_role_id')}
+        value={role.id}
+        type="hidden"
+      />
+      <input name={fieldName('position')} value={idx} type="hidden" />
```

Comment:
Position is indexed from 1, without this, first save saves wrong order as 0 is put at the end, because the library assigns first free position to it.
```suggestion
      <input name={fieldName('position')} value={idx+1} type="hidden" />
```
Or we could directly send position to this Component and assign it as `idx + 1` in the parent :thinking: